From b60e90c43376cdb40897a48226452571f192f1bf Mon Sep 17 00:00:00 2001
From: Alex Deryskyba <alex@codesnake.com>
Date: Thu, 4 Sep 2014 00:28:23 +0300
Subject: [PATCH 06/30] [aml] Update all virtual consoles when changing
 framebuffer geometry after a resolution change

That helps to resolve the issue when framebuffer size is not always correctly change according to the resolution set.
It seems that the issue appeared after updating to the U-Boot bootloader released 2014-08-24 by Amlogic.
---
 xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp |    1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp b/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp
index bf44342..9543119 100644
--- a/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp
@@ -284,6 +284,7 @@ void CEGLNativeTypeAmlogic::SetFramebufferResolution(int width, int height) cons
       vinfo.xres_virtual = 1920;
       vinfo.yres_virtual = 2160;
       vinfo.bits_per_pixel = 32;
+      vinfo.activate = FB_ACTIVATE_ALL;
       ioctl(fd0, FBIOPUT_VSCREENINFO, &vinfo);
     }
     close(fd0);
-- 
1.7.10.4

